src/Store/ExtensionClient.re

```diff
@@ -138,41 +138,29 @@ module ExtensionDocumentSymbolProvider = {
   };
 };
 
-let create = (~extensions, ~setup: Setup.t) => {
+let create =
+    (~config: Feature_Configuration.model, ~extensions, ~setup: Setup.t) => {
   let (stream, dispatch) = Isolinear.Stream.create();
 
-  let manifests =
-    List.map((ext: ExtensionScanner.t) => ext.manifest, extensions);
-
-  let defaults = Configuration.Model.ofExtensions(manifests);
-  let keys = ["reason_language_server.location"];
-
-  let contents =
-    `Assoc([
-      (
-        "reason_language_server",
-        `Assoc([("location", `String(setup.rlsPath))]),
-      ),
-      (
-        "terminal",
-        `Assoc([
-          (
-            "integrated",
-            `Assoc([
-              (
-                "env",
-                `Assoc([
-                  ("windows", `Null),
-                  ("linux", `Null),
-                  ("osx", `Null),
-                ]),
-              ),
-            ]),
-          ),
-        ]),
-      ),
-    ]);
-  let user = Configuration.Model.create(~keys, contents);
+  let defaults =
+    extensions
+    |> List.map(ext =>
+         ext.ExtensionScanner.manifest.contributes.configuration
+       )
+    |> List.map(ExtensionContributions.Configuration.toSettings)
+    |> Config.Settings.unionMany
+    |> Config.Settings.union(Config.Schema.defaults(config.schema))
+    |> Configuration.Model.fromSettings;
+
+  let user =
+    Config.Settings.fromList([
+      ("reason_language_server.location", Json.Encode.string(setup.rlsPath)),
+      ("terminal.integratied.enc.windows", Json.Encode.null),
+      ("terminal.integratied.enc.linux", Json.Encode.null),
+      ("terminal.integratied.enc.osx", Json.Encode.null),
+    ])
```

Comment:
This is one way we can hard-code settings now. Don't know if this is still the way we want to do it though.